crates/core/src/db/datastore/traits.rs

```diff
@@ -322,6 +322,26 @@ pub struct Program {
     pub bytes: Box<[u8]>,
 }
 
+impl Program {
+    /// Create a [`Program`] from its raw bytes.
+    ///
+    /// This computes the hash over `bytes`, so prefer constructing [`Program`]
+    /// directly if the hash is already known.
+    pub fn from_bytes(bytes: impl Into<Box<[u8]>>) -> Self {
+        let bytes = bytes.into();
+        let hash = hash_bytes(&bytes);
+        Self { hash, bytes }
+    }
+
+    /// Create a [`Program`] with no bytes.
+    pub fn empty() -> Self {
```

Comment:
Technically the address of a zero byte program is: `c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470`